android/src/main/java/com/expensify/livemarkdown/MarkdownTextWatcher.java

```diff
@@ -22,17 +22,18 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
 
   @Override
   public void onTextChanged(CharSequence s, int start, int before, int count) {
-    if (mShouldSkip) {
-      return;
-    }
-    if (s instanceof SpannableStringBuilder) {
-      mMarkdownUtils.applyMarkdownFormatting((SpannableStringBuilder) s);
-      mShouldSkip = true;
-    }
+   
   }
 
   @Override
   public void afterTextChanged(Editable editable) {
-    mShouldSkip = false;
+      if (editable instanceof SpannableStringBuilder) {
+          String currentEditable = editable.toString();
+          if (mPreviousEditable != null && mPreviousEditable.equals(currentEditable)) {
+              return;
+          }
+          mMarkdownUtils.applyMarkdownFormatting((SpannableStringBuilder) editable);
+          mPreviousEditable = currentEditable;
```

Comment:
Let's rename `editable` to `text` to avoid confusion (`editable` contains spans and `text` is just a regular String)
```suggestion
          String currentText = editable.toString();
          if (mPreviousText != null && mPreviousText.equals(currentText)) {
              return;
          }
          mMarkdownUtils.applyMarkdownFormatting((SpannableStringBuilder) editable);
          mPreviousText = currentText;
```